src/commands/kv/bucket/upload.rs

```diff
@@ -94,19 +96,31 @@ fn call_put_bulk_api(
     Ok(())
 }
 
-fn filter_unchanged_remote_files(
-    pairs: Vec<KeyValuePair>,
-    exclude_keys: &HashSet<String>,
-) -> Vec<KeyValuePair> {
+fn filter_files(pairs: Vec<KeyValuePair>, already_uploaded: &HashSet<String>) -> Vec<KeyValuePair> {
     let mut filtered_pairs: Vec<KeyValuePair> = Vec::new();
     for pair in pairs {
-        if !exclude_keys.contains(&pair.key) {
+        if !already_uploaded.contains(&pair.key) && !if_ignored_prefix(&pair.key) {
             filtered_pairs.push(pair);
         }
     }
     filtered_pairs
 }
 
+// todo(gabbi): Replace all the logic below with a proper .wignore implementation
+// when possible.
+const KNOWN_UNNECESSARY_PREFIXES: &'static [&str] = &[
+    "node_modules/", // npm vendoring
+    "component---",  // Gatsby sourcemaps
+];
+fn if_ignored_prefix(key: &str) -> bool {
+    for prefix in KNOWN_UNNECESSARY_PREFIXES {
+        if key.starts_with(prefix) {
```

Comment:
for this first pass w/ just node_modules, should we ignore any path that includes node_modules?

Reply:
Hmm, where else would node modules live? I'm not opposed to it, it just seems like unorthodox ignoring behavior (it usually follows prefixes, not the presence of a string)

Reply:
i noticed as i was developing https://github.com/cloudflare/wrangler/pull/708 that the key itself still includes the directory name, so this... might not work? or if it does i am confused as to why?

Reply:
turns out i broke this.

Reply:
I added more finessed logic for looking at whether a `node_module` subdirectory existed anywhere within a bucket :) So @ashleymichal we should have your suggested functionality ready to go!